frontends/p4/sideEffects.h

```diff
@@ -28,7 +28,7 @@ limitations under the License.
 namespace P4 {
 
 /// Checks to see whether an IR node includes a table.apply() sub-expression
-class HasTableApply : public Inspector {
+class HasTableApply : public Inspector, public ResolutionContext {
```

Comment:
If you make `HasTableApply` a `ResolutionContext`, you don't need the `refMap` field in it any more -- you just use `this`.  However, as `HasTableApply` is only ever called on expressions, there won't be any additional scopes to look things up in within it's visit traversal -- the innermost scope will be the same as the caller.  So I don't see how this affects anything.

Reply:
It is still called with refmap somewhere in the midend. And I do not want to trigger set of changes there.

Reply:
And it does affect exactly due to outlined case, see https://github.com/p4lang/p4c/blob/676bce5323c11f27f858a841f25ea8ca3c86b03f/frontends/common/resolveReferences/resolveReferences.cpp#L183 – it is only be able to resolve things on the context.

So, `KeySideEffects` cannot resolve overload, it it "further down".

Reply:
Please add a comment here that this needs to be `ResolutionContext` so that if we are working without a `P4::ReferenceMap` we have the call in context for the sake of `MethodInstance::resolve`.